taichi/program/kernel.cpp

```diff
@@ -75,8 +75,7 @@ void Kernel::lower(bool lower_access) {  // TODO: is a "Lowerer" class necessary
         (is_evaluator && !config.print_evaluator_ir))
       verbose = false;
 
-    // For now we only enable TLS on CPUs
-    bool make_thread_local = arch_is_cpu(arch);
+    bool make_thread_local = true;
```

Comment:
Only CPU or CUDA?

Reply:
Yeah now we support both CPU and CUDA.

Reply:
Ah, i meant to ask if this could break on other platforms like GL and Metal..

Reply:
Oh we have this line above :-)
```
if (arch_is_cpu(arch) || arch == Arch::cuda) {
```